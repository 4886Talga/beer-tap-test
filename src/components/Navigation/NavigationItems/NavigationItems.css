.NavigationItems {
    margin: 0;
    padding: 0;
    list-style: none;
    display: flex;
    flex-flow: column;
    align-items: center;   
    height: 100%;
}


@media(min-width: 500px) {
    .NavigationItems {
        flex-flow: row;
    }
    .UserName {
        border: solid gray 1px;
        color: #333;
        padding: 2px;
        margin-top: -3px;
    }
    .SignOut {
        color: white;
    }
}