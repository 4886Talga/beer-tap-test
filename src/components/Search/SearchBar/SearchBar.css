.SerchHolder {
    position: absolute;
    top: 56px;
    left: 0px;
    z-index: 10;
    width: 100%;
    padding-right: 0px;
    transition: all 0.3s ease-out;
}
.Button {
    position: absolute;
    background: transparent;
    color: #ad9d7a;
    border: 0px none;
    height: 50px;
    padding: 0 30 0 0;
    width: auto;
    right: 10px;
    top: 0px;
    margin: 0px;
}
.Space{
    margin-left: 10px;
    vertical-align: 5px;
}
.SearchInput {
    width: 100%;
    box-shadow: none;
    margin: 0px;
    padding: 0px 150px 0px 15px;
    box-sizing: border-box;
    border: 1px solid #ad9d7a;
    font-family: "lato",serif;
    height: 50px;
    font-size: 16px;
}
