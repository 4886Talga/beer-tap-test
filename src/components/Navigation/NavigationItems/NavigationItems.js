import React, { Component } from 'react';
import {FaSearch} from 'react-icons/lib/fa';
import './NavigationItems.css';
import NavigationItem from '../NavigationItem/NavigationItem';
import SearchBar from '../../Search/SearchBar/SearchBar';
import Aux from '../../../extra/extra';


class NavigationItems extends Component {
    state = {
        show: 'none'
    }
    showSearchBar = () => {        
		this.state.show === 'none' ? this.setState({show:'inline'}) : this.setState({show:'none'});
	}
    render() {
        
        return(
            <Aux>
                <ul className="NavigationItems">
                    <li><div className="SearchBar" onClick={this.showSearchBar}><FaSearch /></div></li>
                    <NavigationItem link="/" exact drawerToggleClicked={this.props.drawerToggleClicked}>HOME</NavigationItem>
                    <NavigationItem link="/gallery" drawerToggleClicked={this.props.drawerToggleClicked}>BEERS</NavigationItem>
                    { !this.props.user && (<NavigationItem link="/login" drawerToggleClicked={this.props.drawerToggleClicked}>LOGIN</NavigationItem>)}
                    { !this.props.user && (<NavigationItem link="/createaccount" drawerToggleClicked={this.props.drawerToggleClicked}>REGISTER</NavigationItem>)}
                    { this.props.user && (<NavigationItem link="/logout" drawerToggleClicked={this.props.drawerToggleClicked}>LOGOUT</NavigationItem>)}
                    { this.props.user && (<NavigationItem link="/dashboard" drawerToggleClicked={this.props.drawerToggleClicked}>MINA SIDOR</NavigationItem>)}
                    { this.props.user && (<li><div className="UserName">{this.props.user.displayName}</div></li>)}
                </ul>
                <SearchBar show={this.state.show} />
            </Aux>
        );
    }
}
export default NavigationItems;