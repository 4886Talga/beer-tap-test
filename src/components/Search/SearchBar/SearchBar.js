import React, { Component } from 'react';
import {FaSearch} from 'react-icons/lib/fa';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import './SearchBar.css';

class SearchBar extends Component {
	
	state = { 
		term: ''
	};
	
		
	handleFormSubmit = (e) => {
		//e.preventDefault();
		console.log(this.state.term);
		this.handleClearForm(e);
	  }
	
	handleClearForm = (e) => {
		//e.preventDefault();
		this.setState({term: ''});
		e.target.value = '';
	}

	
	render() {
		
		return (
			<form>
			<div className="SerchHolder" style={{display: this.props.show}}>
					<input className="SearchInput" 
						value={this.state.term}
						onChange={(event) => this.onInputChange(event.target.value)} 
						placeholder={'Search for BrewDogs beer, malt, hops, yeast, food...'} />
					<button className="Button">
						<Link to={{pathname:`/searchresault`, state: {searchterm:this.state.term} }}
							 onClick={this.handleFormSubmit}>SEARCH<span className="Space"><FaSearch /></span>
						</Link>
					</button>
			</div>
			
			</form>
		)
			
	}
	
		
	onInputChange(term) {
		this.setState({term});
	
	}
	
}

SearchBar.propTypes = {
		show: PropTypes.string.isRequired
	};

export default SearchBar;