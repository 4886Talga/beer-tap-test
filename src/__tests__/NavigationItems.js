/**
 * @jest-environment jsdom
 */
import React from 'react';
import {FaSearch} from 'react-icons/lib/fa';
import { shallow, mount, render } from 'enzyme';
import NavigationItems from '../components/Navigation/NavigationItems/NavigationItems';
import NavigationItem from '../components/Navigation/NavigationItem/NavigationItem';
import SearchBar from '../components/Search/SearchBar/SearchBar';


describe('testing <NavigationItems />', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<NavigationItems />);
    });
    it('should render one <FaSearch /> components', () => {
         expect(wrapper.find(FaSearch)).toHaveLength(1);
      });
    
    it('should render four <NavigationItem /> elements if not authenticated', () => {
        expect(wrapper.find(NavigationItem)).toHaveLength(4);
    });
    it('should render four <NavigationItem /> elements if authenticated', () => {
        // wrapper = shallow(<NavigationItems user />);
        wrapper.setProps({user: true});
        expect(wrapper.find(NavigationItem)).toHaveLength(4);
    });

    it('should have an exact logout button', () => {
        wrapper.setProps({user: true});
        expect(wrapper.contains(<NavigationItem link="/logout">LOGOUT</NavigationItem>)).toEqual(true);
    });
    it('should have an <li> with class userName if authenticated', () => {
        wrapper.setProps({user: true});
        expect(wrapper.find('.UserName')).toHaveLength(1);
    });

    it('should render correctly', () => {
        
        expect(wrapper).toMatchSnapshot();
      });

   
});
describe('simulates click events', () => { 
    it('should show SearchBar onClick', () => {
        const wrapper = shallow(<NavigationItems />);
        wrapper.setProps({user: true});
        expect(wrapper.state('show')).toEqual('none');
        wrapper.find('.SearchBar').simulate("click");
        expect(wrapper.state('show')).toEqual('inline');
        wrapper.find('.SearchBar').simulate("click");
        expect(wrapper.state('show')).not.toBe('inline');
        expect(wrapper.state('show')).toEqual('none');
      });
});
describe('<SearchBar />', () => { 
    it('should load component and check if props sends', () => {
        const wrapper = shallow(<NavigationItems />);
        const { show } = wrapper.state();
        const searchBar = shallow( <SearchBar show={show} />);
        expect(searchBar.instance().props.show).toEqual('none');       
    });
});
