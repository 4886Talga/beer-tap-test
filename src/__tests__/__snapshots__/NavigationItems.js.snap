// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing <NavigationItems /> should render correctly 1`] = `
<aux>
  <ul
    className="NavigationItems"
  >
    <li>
      <div
        className="SearchBar"
        onClick={[Function]}
      >
        <FaSearch />
      </div>
    </li>
    <navigationItem
      exact={true}
      link="/"
    >
      HOME
    </navigationItem>
    <navigationItem
      link="/gallery"
    >
      BEERS
    </navigationItem>
    <navigationItem
      link="/login"
    >
      LOGIN
    </navigationItem>
    <navigationItem
      link="/createaccount"
    >
      REGISTER
    </navigationItem>
  </ul>
  <SearchBar
    show="none"
  />
</aux>
`;
